runtime/health.go

```diff
@@ -58,5 +50,183 @@ func (r *Runtime) Health(ctx context.Context) (*Health, error) {
 }
 
 func (r *Runtime) InstanceHealth(ctx context.Context, instanceID string) (*InstanceHealth, error) {
-	return r.registryCache.instanceHealth(ctx, instanceID)
+	res := &InstanceHealth{}
+	// check repo error
+	err := r.pingRepo(ctx, instanceID)
+	if err != nil {
+		res.Repo = err.Error()
+	}
+
+	ctrl, err := r.Controller(ctx, instanceID)
+	if err != nil {
+		res.Controller = err.Error()
+		return res, nil
+	}
+
+	parser, err := ctrl.Get(ctx, GlobalProjectParserName, false)
+	if err != nil {
+		return nil, err
+	}
+	res.ParseErrCount = len(parser.GetProjectParser().State.ParseErrors)
+
+	cachedHealth, _ := r.cachedInstanceHealth(ctx, ctrl.InstanceID, ctrl.catalog.version)
+	// set to true if any of the olap engines can be scaled to zero
+	var canScaleToZero bool
+
+	// check OLAP error
+	olap, release, err := r.OLAP(ctx, instanceID, "")
+	if err != nil {
+		res.OLAP = err.Error()
+	} else {
+		mayBeScaledToZero := olap.MayBeScaledToZero(ctx)
+		canScaleToZero = canScaleToZero || mayBeScaledToZero
+		if cachedHealth != nil && mayBeScaledToZero {
+			res.OLAP = cachedHealth.OLAP
+		} else {
+			err = r.pingOLAP(ctx, olap)
+			if err != nil {
+				res.OLAP = err.Error()
+			}
+		}
+		release()
+	}
+
+	// run queries against metrics views
+	resources, err := ctrl.List(ctx, ResourceKindMetricsView, "", false)
+	if err != nil {
+		return nil, err
+	}
+	res.MetricsViews = make(map[string]metricsViewHealth, len(resources))
+	for _, mv := range resources {
+		if mv.GetMetricsView().State.ValidSpec == nil {
+			if mv.Meta.ReconcileError != "" {
+				res.ReconcileErrCount++
```

Comment:
This only captures reconcile error counts for metrics views, not other resources?

Reply:
I thought we were only interested in the metrics views and not other resources. We can add all resources.